filesystems/tests/common.py

```diff
@@ -145,6 +149,47 @@ def test_open_append_non_existing_nested_file(self):
             )
         )
 
+    def _writelines(self, to_write, newline, mode, text):
+        fs = self.FS()
+        tempdir = fs.temporary_directory()
+        self.addCleanup(fs.remove, tempdir)
+
+        with fs.open(tempdir.descendant("unittesting"), mode) as f:
+            f.writelines(line + newline for line in to_write.splitlines())
+
+        with fs.open(tempdir.descendant("unittesting")) as g:
+            self.assertEqual(g.read(), text)
+
+    def test_writelines_bytes(self):
+        text = u"some\nthings!\n"
+
+        self._writelines(
```

Comment:
This test looks a bit bare, like over-concise -- I think you generally want to be able to stare at a test and at least see whether it has the 3 expected stages (set up, exercise, assert), but here even the assert is hidden behind a bit of a non-obvious name (`_writelines`)

So while I definitely like factoring out common stuff, it seems like possibly there's a middle stage here that isn't as drastically factored out. I think having to do the same 3 lines to create a tempdir to work in is pretty annoying myself, but unittest makes it annoying to do that setup on only some subset of tests, but possibly we should consider some of the things that can be done about that...

Reply:
I had pytest parametrization in mind.  Multiple results from 'one function'.  A quick search turned up this for unittest.  I'll look around and see if i can't find something better.

Reply:
I guess I should ask, do you have an attachment to the trial/unittest setup?  I can't really compare to other systems as I've only used pytest for my own stuff.  There are aspects I'm not a huge fan of but it does provide parametrization (covers this case) and fixtures (would cover the general 3-line tempdir repetition).

I'm not saying filesystems should switch to pytest, just that I (sort of) know how to use it unlike unittest.  :|  and it _might_ be better.

Reply:
Yeah long story :), in general mostly I don't like pytest because it built a pytest ecosystem instead of building an ecosystem that could be more widely used -- I'm not vehemently opposed to using it at this point, the battle is mostly lost, so maybe we switch to it, but just to point you in the incremental direction here, for fixtures, one would use the [fixtures](https://pypi.org/project/fixtures/) library, or just write a 3 line helper, or split this into 2 test cases and just use `setUp`, whichever, and for parametrization, you'd use [testscenarios](https://pypi.org/project/testscenarios/).

I'd say I'm OK with whatever you choose though the smaller the change the easier it'll be for me to review it quickly.